packages/insomnia/src/ui/components/request-url-bar.tsx

```
@@ -51,18 +53,25 @@ export const RequestUrlBar = forwardRef<RequestUrlBarHandle, Props>(({
   onPaste,
 }, ref) => {
   const [searchParams, setSearchParams] = useSearchParams();
+  const [showEnvVariableMissingModal, setShowEnvVariableMissingModal] = useState(false);
+  const [missingKey, setMissingKey] = useState('');
   if (searchParams.has('error')) {
-    showAlert({
-      title: 'Unexpected Request Failure',
-      message: (
-        <div>
-          <p>The request failed due to an unhandled error:</p>
-          <code className="wide selectable">
-            <pre>{searchParams.get('error')}</pre>
-          </code>
-        </div>
-      ),
-    });
+    if (searchParams.has('envVariableMissing')) {
+      setShowEnvVariableMissingModal(true);
+      setMissingKey(searchParams.get('missingKey') || '');
```

When the missing key is `''` or falsy values, probably we could skip showing modal in this case? or throw an error if it is unexpected.